core/src/main/java/hudson/util/ProcessTree.java

```diff
@@ -788,6 +788,13 @@ protected OSProcess createProcess(final int pid) throws IOException {
             private static final byte PR_MODEL_LP64 = 2;
 
             /*
+             * An arbitrary upper-limit on how many characters readLine() will
+             * try reading before giving up. This avoids having readLine() loop
+             * over the entire process address space if this class has bugs.
+             */
+            private static final int LINE_LENGTH_LIMIT = 10000;
```

Comment:
Good practice is to make it a SystemProperty in order to allow managing it from the code. Maybe not required here since the limit is big enough for realistic use-cases